x/gamm/keeper/genesis.go

```diff
@@ -36,9 +36,9 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState, unpack
 	k.setTotalLiquidity(ctx, liquidity)
 
 	if genState.MigrationRecords == nil {
-		k.OverwriteMigrationRecords(ctx, types.MigrationRecords{})
+		k.OverwriteMigrationRecordsNoRedirect(ctx, types.MigrationRecords{})
 	} else {
-		k.OverwriteMigrationRecords(ctx, *genState.MigrationRecords)
+		k.OverwriteMigrationRecordsNoRedirect(ctx, *genState.MigrationRecords)
```

Comment:
Note to reviewer: we need a separate NoRedirect method to strictly be used in the genesis initialization. This is because, when we run export genesis, we expect the distribution records to have already been properly handled. 

If we try to run distribution record logic within the gamm module in init genesis, this does not work, since the incentives module must get initialized post gamm module initialization (pools must exist prior to setting up the distribution records for them).

I am fairly confident with this logic, but just wanted to explain that in depth here since it wasn't straight forward initially when implementing it.